### DISCUSSION TITLE
question about location of cached models

### DISCUSSION THREAD

**Discussion by justADeni** [rupeshs/fastsdcpu#102] [Q&A] [answered]
When I downloaded fastsdcpu, I made sure to put it on my 2TB drive because there's plenty of space. Yet to my horror I checked that all models got downloaded cached to ./root/.cache/huggingface/hub/<model>/blobs/<files> on my boot drive (which is comparatively small and full.

Is there any way to change where the cached models go?

**Comment by gustrd** [comment]
Are you running as root? It uses the user folder.

**Reply by justADeni** [reply]
I may have run it with sudo, yes. But even if I didn't, you're saying it would still end up in my user folder, i.e. on my boot drive? Is there not an option to make it use the location where it's installed?

**Reply by gustrd** [reply, marked as answer]
Maybe you can try this: https://stackoverflow.com/questions/63312859/how-to-change-huggingface-transformers-default-cache-directory

**Reply by justADeni** [reply]
thank you, that is super helpful!

**Reply by gonzaloonofreHV** [reply]
Hi, I guess we have to write:
 import os
os.environ['TRANSFORMERS_CACHE'] = '/blabla/cache/'
but where? I mean, which file do we add that to? Sorry, I am new at these things, it must be obvious to you.

**Reply by gustrd** [reply]
What's your Operational System?

**Reply by gonzaloonofreHV** [reply]
Windows

**Reply by gustrd** [reply]
This guide can help you:

https://phoenixnap.com/kb/windows-set-environment-variable

**Reply by tukangcode** [reply]
hi sorry i have same issue to but i totaly noob on this, is there simple way to change location aside i need manualy do it ? like it being config before install

**Reply by tukangcode** [reply]
I really confused with method above maybe video can help. I found some guide here for windows but this not work for me https://github.com/kohya-ss/sd-scripts/issues/384 anyone can figure out this script code they use

``@echo off

:: Activate the virtual environment
call .\venv\Scripts\activate.bat

:: Validate the requirements and store the exit code
python.exe .\tools\validate_requirements.py

set HF_HOME=D:\SDCache

:: If the exit code is 0, run the kohya_gui.py script with the command-line arguments
if %errorlevel% equ 0 (
    python.exe kohya_gui.py %*
)

**Reply by tukangcode** [reply]
Hi @justADeni how you do it?

**Reply by rupeshs** [reply]
@tukangcode try this for for winodws `set HF_HOME=D:/mycache` .

**Comment by gonzaloonofreHV** [comment]
I did change the location but havent been able to make the program work
since, so...

**Reply by rupeshs** [reply]
try this for for winodws `set HF_HOME=D:/mycache`

**Reply by gonzaloonofreHV** [reply]
Thanks, that IS what I did. i can not figure out why It IS not working. Sorry about caps, typing on phone.

**Reply by rupeshs** [reply]
Try a fresh installation

**Comment by rupeshs** [comment]
To change the cache path set **HF_HOME** environment variable.
For Winodws `set HF_HOME=D:/mycache`
For Linux/Mac `export HF_HOME=/home/mycache`
Also, you can add this to start.bat or start.sh file.
This should work!
